src/builder.rs

```diff
@@ -213,30 +216,59 @@ impl<'ctx> Builder<'ctx> {
     ///
     /// builder.position_at_end(entry);
     ///
-    /// let ret_val = builder.build_call(fn_type, fn_value, &[i32_arg.into(), md_string.into()], "call")
+    /// let ret_val = builder.build_call(fn_value, &[i32_arg.into(), md_string.into()], "call")
     ///     .try_as_basic_value()
     ///     .left()
     ///     .unwrap();
     ///
     /// builder.build_return(Some(&ret_val));
     /// ```
     #[llvm_versions(15.0..=latest)]
-    pub fn build_call<F>(
+    pub fn build_call(
         &self,
-        fn_ty: FunctionType<'ctx>,
-        function: F,
+        function: FunctionValue<'ctx>,
         args: &[BasicMetadataValueEnum<'ctx>],
         name: &str,
-    ) -> CallSiteValue<'ctx>
-    where
-        F: Into<CallableValue<'ctx>>,
-    {
-        let callable_value = function.into();
-        let fn_ty_ref = fn_ty.as_type_ref();
-        let fn_val_ref = callable_value.as_value_ref();
+    ) -> CallSiteValue<'ctx> {
+        self.build_direct_call(function, args, name)
+    }
+
+    #[llvm_versions(15.0..=latest)]
+    pub fn build_direct_call(
+        &self,
+        function: FunctionValue<'ctx>,
+        args: &[BasicMetadataValueEnum<'ctx>],
+        name: &str,
+    ) -> CallSiteValue<'ctx> {
+        self.build_call_help(function.get_type(), function.as_value_ref(), args, name)
+    }
 
+    #[llvm_versions(15.0..=latest)]
+    pub fn build_indirect_call(
```

Comment:
Do I understand correctly that this function can be used everywhere, for both global function values and generic function pointers?

Reply:
you could turn a function into a global pointer and then use that with this function, yes. But then you have to supply the function type explicitly.

Reply:
Cool, that's what I need.